src/platformAccessory.ts

```diff
@@ -340,10 +347,20 @@ export class HomebridgeMagichomeDynamicPlatformAccessory {
     this.service.updateCharacteristic(this.platform.Characteristic.On,  this.lightState.isOn);
     this.service.updateCharacteristic(this.platform.Characteristic.Hue, this.lightState.HSL.hue);
     this.service.updateCharacteristic(this.platform.Characteristic.Saturation, this.lightState.HSL.saturation);
-    if(this.lightState.HSL.luminance > 0 && this.lightState.isOn){
+    if (
+      !this.myDevice.lightParameters.hasCCT &&
+      this.lightState.HSL.luminance > 0 &&
+      this.lightState.isOn
+    ) {
       this.updateLocalBrightness(this.lightState.HSL.luminance * 2);
```

Comment:
It's unclear to me why the local brightness value gets updated here, but this seems to cause HomeKit to show the wrong color value when navigating away and back to the accessory settings, if the color value was set to a color temperature rather than RGB.